src/detectLaunchConfigurationChanges.ts

```diff
@@ -8,15 +8,19 @@ interface PromptRestartParams {
 
 export function detectLaunchConfigurationChanges(
   workspace: Workspace,
-  promptRestart: (params: PromptRestartParams) => Thenable<void>
+  promptRestart: (params: PromptRestartParams) => Thenable<void>,
+  additionalTriggers?: string[]
 ): void {
   workspace.onDidChangeConfiguration(e => {
-    const promptRestartKeys = [
+    const initialRestartKeys = [
       "serverVersion",
       "serverProperties",
       "javaHome",
-      "customRepositories"
+      "customRepositories",
     ];
+    const promptRestartKeys = additionalTriggers
+      ? [...initialRestartKeys, ...additionalTriggers]
+      : initialRestartKeys
```

Comment:
what about defaulting the param to [] and avoiding this ternary?

Reply:
Ah, good call. I switched this. 👍